cf_units/cf_units.py

```diff
@@ -830,7 +833,47 @@ class Unit(util._OrderedHashable):
     This class also supports time and calendar defintion and manipulation.
 
     """
-    # Declare the attribute names relevant to the _OrderedHashable behaviour.
+    def _init_from_tuple(self, values):
+        for name, value in zip(self._names, values):
+            object.__setattr__(self, name, value)
+
+    def _as_tuple(self):
+        return tuple(getattr(self, name) for name in self._names)
+
+    # Provide hash semantics
+
+    def _identity(self):
+        return self._as_tuple()
+
+    def __hash__(self):
+        return hash(self._identity())
+
+    def __eq__(self, other):
+        return (isinstance(other, type(self)) and
+                self._identity() == other._identity())
+
+    def __ne__(self, other):
+        # Since we've defined __eq__ we should also define __ne__.
+        return not self == other
+
+    # Provide default ordering semantics
+
+    def __lt__(self, other):
+        a, b = self._identity(), other._identity()
+        return (a > b) - (a < b)
```

Comment:
Aha, makes sense!  This is what I get for copying-and-pasting without
thinking!

Thanks!

On Tue, May 5, 2015 at 6:17 PM, Elliott Sales de Andrade <
notifications@github.com> wrote:

> In cf_units/cf_units.py
> <https://github.com/SciTools/cf_units/pull/22#discussion_r29714064>:
>
> > +    def __hash__(self):
> > +        return hash(self._identity())
> > +
> > +    def __eq__(self, other):
> > +        return (isinstance(other, type(self)) and
> > +                self._identity() == other._identity())
> > +
> > +    def __ne__(self, other):
> > +        # Since we've defined __eq__ we should also define __ne__.
> > +        return not self == other
> > +
> > +    # Provide default ordering semantics
> > +
> > +    def __lt__(self, other):
> > +        a, b = self._identity(), other._identity()
> > +        return (a > b) - (a < b)
>
> Well, less-than would just be return a < b, no?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciTools/cf_units/pull/22/files#r29714064>.
>